src/lightning/data/streaming/reader.py

```diff
@@ -47,18 +49,41 @@ def add(self, chunk_indices: List[int]) -> None:
                 if chunk_indice not in self._chunks_index_to_be_processed:
                     self._chunks_index_to_be_processed.append(chunk_indice)
 
+    def delete(self, chunk_indices: List[int]) -> None:
+        """Receive the list of the chunk indices to download for the current epoch."""
+        with self._lock:
+            for chunk_indice in chunk_indices:
+                if chunk_indice not in self._chunks_index_to_be_deleted:
+                    self._chunks_index_to_be_deleted.append(chunk_indice)
+
+    def _delete(self, chunk_index: int) -> None:
+        chunk_filepath, begin, _ = self._config[ChunkedIndex(index=-1, chunk_index=chunk_index)]
+
+        if os.path.exists(chunk_filepath):
+            os.remove(chunk_filepath)
+
     def run(self) -> None:
         while True:
             with self._lock:
+                if len(self._chunks_index_to_be_processed) == 0 and len(self._chunks_index_to_be_deleted) == 0:
+                    sleep(0.0001)
+                    continue
+
+                # Delete the chunks if we are missing disk space.
+                # Check every 10 items to avoid losing too much time on the subprocess
+                if len(self._chunks_index_to_be_deleted) > 10:
+                    if _get_available_amount_of_bytes() < _MINIMUM_DISK_SIZE:
```

Comment:
I was expecting something like a `MAXIMUM_CACHE_SIZE` here.

Being able to specify `MAXIMUM_CACHES_SIZE` instead of `MINIMUM_DISK_SIZE` would lead to a more predictable system IMO. The way it is right now, how much I'm caching will depend on what happens outside of the system (e.g. how many checkpoints I'm saving) and it might be surprising.

In the case we set maximum cache size, I would suggest having a `max_cache_size` as a parameter of the streaming dataset, rather than a global.

In any case, I feel the name `MINIMUM_DISK_SIZE` is a bit unclear.

Reply:
I think the main challenge with the max approach comes when using combined dataset. They are all adding to themselves and now the user needs to adapt its code.

Reply:
But I can expose this through the StreamingDataset as you suggested.

Reply:
I see, what about having a single size for the combination?
I feel that that is a problem no matter what, we're just kind of deferring it to where it breaks.

I don't want this to block us right now, just something to think about.

Reply:
Let's get @awaelchli opinion, maybe `max_cache_size ` resonable more with him.

Reply:
@lantiga I moved to `max_cache_size` as you suggested. Yes, I think we would need a `ingle size for the combination` but this is probably fine.

Reply:
Leaning towards @lantiga's suggestion. Making a good default of `max_cache_size` sounds intuitive, like 200GB total for all processes or something in that range. Since we delete the chunks, we should have enough room to play even if our default is conservative, especially on a deep learning purposed system there will be plenty of space for a streaming dataset to operate.

Reply:
Done. I have put 200GB as a default